src/components/hv-progress-bar/index.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+
+/**
+ * Copyright (c) Garuda Labs, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import * as Namespaces from 'hyperview/src/services/namespaces';
+import React, { PureComponent } from 'react';
+import { createProps } from 'hyperview/src/services';
+import type { HvComponentProps } from 'hyperview/src/types';
+import { Animated, View } from 'react-native';
+import { LOCAL_NAME } from 'hyperview/src/types';
+
+export default class HvImage extends PureComponent<HvComponentProps> {
+  static namespaceURI = Namespaces.HYPERVIEW;
+  static localName = LOCAL_NAME.PROGRESS_BAR;
+  static localNameAliases = [];
+  props: HvComponentProps;
+
+  constructor(props) {
+    super(props);
+    const element = this.props.element;
+    const startPercent: ?number = this.getStartPercent(element);
+    const initialPercent: number =
+      startPercent === null ? this.getPercent(element) || 0 : startPercent;
+    this.state = {
+      barPercent: new Animated.Value(initialPercent),
+    };
+  }
+
+  getStartPercent = (element): ?number => {
```

Comment:
These questions apply to `getStartPercent`, `getPercent` and `getMaxDuration`:
- can `element` be flow-typed?
- shouldn't the if statement be `if (!startPercent)` ? Are we omitting `undefined` value for a specific reason?

Reply:
Yes, I will add full Flow types in the final PR, I didn't focus on it too much yet in case we wanted to go in a different direction. Same for handling falsy values.